server/bell_oauth_profile.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var log = require('./logger')('server.bell.profile');
+var db = require('./db/db');
+var config = require('./config');
+
+// this is the custom provider profile function used by Bell to allow us
+// to convert oauth tokens into profile information.
+function profile (credentials, params, get, profileCb) {
+  log.verbose('obtained oauth tokens: ' + JSON.stringify(credentials));
+  var headers = { headers: {'authorization': 'Bearer ' + params.access_token} };
+  get(config.get('server.oauth.profileEndpoint'), headers, function(data) {
+    // Bell returns the parsed data and handles errors internally
+    log.verbose('exchanged tokens for profile data:' + JSON.stringify(data));
+    // TODO use Joi to validate `data` before sending to DB
+    db.getUserById(data.uid, function (err, user) {
+      if (err) {
+        log.warn('getUserById failed: ' + err);
```

Comment:
@seanmonstar Cool! Thanks for the feedback. I haven't had a chance to run this through Heka yet, great to catch these problems before we go to production.

Does the `var log = require('./logger')('server.bell.profile');` on line 7 handle the initial string with no spaces bit, or is that a different part of the logline namespacing?